### DISCUSSION TITLE
Doubts on internationalization and overall customization of the editor and components

### DISCUSSION THREAD

**Discussion by sandenson** [GrapesJS/grapesjs#3818] [Q&A]
Hello! I've just recently started using GrapesJS along with a team of fellow developers, and we've all been struggling with documentation and finding information.

**Edit: I've since managed to resolve a few of these issues by myself, but if there are better solutions, please let me know.**

The main thing I've been working on is with internationalization of the editor and its components. I was pretty pleased to find out that GrapesJS has built-in internationalization with i18n. I've downloaded the locales from the repository and managed to get most things done, but couldn't figure out how to do some things, and I'm not sure if I can do some of them with the means I'm using, either.

I have three major doubts, at the moment:

1. **(Resolved by myself)** Internationalization of plugin component's names when hovering over them in the editor.
    
    I've noticed that the components are tracked by the internalization features through the `data-gjs-type` property found in the main div element of the components. Taking the Body component as an example, which represents the background of the editor, and has the "wrapper" value in the `data-gjs-type`. In Portuguese, "Body" translates to "Corpo". Here we can see it at work.
![image](https://user-images.githubusercontent.com/29125169/134679033-3575c7a5-bd07-466b-8825-c51351440092.png) ![image](https://user-images.githubusercontent.com/29125169/134679449-608c58f5-19af-4942-9f1d-c8876c974ecf.png) ![image](https://user-images.githubusercontent.com/29125169/134679811-3b2e7e17-ef87-4f64-9984-dd7183b3afc4.png)

    This makes for a problem with plugin components. Most of them have the `data-gjs-type` property set as `default`, while their actual names are within their respective classlists. Take the "1 Column" component from the basic plugin, for example. It is made out of a Cell component inside of a Row component. Both the "Row" and "Cell" types are translated within the locale object, but they aren't translated when the editor is run.
![image](https://user-images.githubusercontent.com/29125169/134681093-9ea9e549-b0bb-4763-8a89-dba3d6747225.png) ![image](https://user-images.githubusercontent.com/29125169/134681188-49bbb3a2-be42-4be1-9411-44ac82ed4daa.png) ![image](https://user-images.githubusercontent.com/29125169/134680802-1af001ff-478f-432b-9a20-42a9da4b5d91.png) ![image](https://user-images.githubusercontent.com/29125169/134681009-3b4cc634-2ae1-4d01-94f3-ad42be96f9f9.png)

    I then tried to manually change either the `data-gjs-type` values or the hover text of each component, but couldn't figure out how to do that, either.

   **Edit: Upon closer inspection, I've noticed that some components which ***do*** have values other than default for their `data-gjs-type`, such as `textarea`, `input` and `label`, and have been set up in the locale file still aren't translated, which makes me unsure about how the internationalization features work.**

    **Edit 2: I've been fiddling with the Form component from the Forms plugin and even when I manually insert the `data-gjs-type` attribute in the children components, the DOM hover tooltip still doesn't show up, further debunking my former theory.**
![image](https://user-images.githubusercontent.com/29125169/134728584-62c41852-17e3-4389-815e-957f4065adb7.png)
    (The t() functions are from the i18n module)

    **Edit 3 (problem resolved): I've just found out about the `data-gjs-custom-name` property, which is what I was looking for.**
2. Internationalize items of selectors in the UI

    This isn't a major concern for what I'm working on, but it'd be great if I could do it. I was wondering if there's a way to translate these (I'm using the font-weight setting as an example):
![image](https://user-images.githubusercontent.com/29125169/134684547-efdc9620-af62-45a9-85f9-6339e398300e.png)

    If someone could help me with this, I'd appreciate it very much. Thank you in advance.

    **Edit: By the way, I'm working with ReactJS and Typescript. I don't know if that's an important piece of information, but I figure I should share it.**

    Edit: https://github.com/artf/grapesjs/issues/2515#issuecomment-944936026

    There you go, @artf, have your discussion.

**Comment by ronaldohoch** [comment]
I would like to have the possibility to translate the options from UI too.
